lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -276,7 +276,13 @@ def main():
     # find the group if present
     group = None
     groups = {}
-    for curGroup in ec2.get_all_security_groups():
+
+    try:
+        security_groups = ec2.get_all_security_groups()
+    except Exception as e:
```

Comment:
@s-hertel I'm happy to catch a more specific exception.  Question: I have gotten timeouts on this call before, I'm not sure that boto will wrap those in EC2ResponseError.  The documentation is kind of sparse and looking around I see others handling both EC2ResponseError and other exceptions.  LMK what you think? Thanks!